MAINT: Prefer PySide6 in testing

larsoner opened the pull request:
~~I think we should cut a release. Given that this will be the first to support Qt6, I bumped to 0.3.0 (last release was 0.2.6).~~

~~@agramfort or @hoechenberger feel free to merge and I'll cut a release~~

Uses PySide6 as our preferred framework in testing. We can't change `environment.yml` until conda-forge manages to release PySide6, which will require Qt6 first. So it might be a while!

Todo

- [x] Fix bug with mne_dspm example (error, or hanging ref?)

larsoner commented:
Argh, I meant for that to be the text for the mne-qt-browser PR I'm about to open. This PR might require a 0.3.0 release, we'll see!

larsoner commented:
PySide6's wheel does not work on 18.04 LTS, and switching to 20.04 LTS we get volume rendering bugs from MESA

| main | PR |
| -- | -- |
![image](https://user-images.githubusercontent.com/2365790/162840872-5b792ddb-6d0f-42ca-bbae-17b985a5f0e8.png) | ![image](https://user-images.githubusercontent.com/2365790/162840820-68bfc7e4-a641-4666-8315-011f78b13d6c.png)

I'll try PyQt6 on CircleCI + Ubuntu 18.04. That might be a way forward at least...

larsoner commented:
Okay this one is good to go. We still use PyQt5 on CircleCI because PySide6 uses too new an ABI for 18.04, and we can't move away from 18.04 until https://gitlab.freedesktop.org/mesa/mesa/-/issues/6309 is sorted out

larsoner commented on mne/viz/_3d.py:
```diff
@@ -3292,24 +3292,28 @@ def plot_brain_colorbar(ax, clim, colormap='auto', transparent=True,
+    antialias: Optional[bool]
+    depth_peeling: Optional[bool]
+    smooth_shading: Optional[bool]
+    multi_samples: Optional[int | None]
```
```suggestion
    multi_samples: Optional[int]
```

larsoner commented:
Now the problem with PySide6 on CircleCI is a remaining reference:

https://app.circleci.com/pipelines/github/mne-tools/mne-python/13724/workflows/cffe0e33-f535-43fc-8b0f-1217939638a8/jobs/43496

I'll take care of that in a separate PR, though. Ready for review/merge from my end!

hoechenberger commented on azure-pipelines.yml:
```diff
@@ -257,9 +257,9 @@ stages:
+        3.9 pip:
           TEST_MODE: 'pip'
-          PYTHON_VERSION: '3.7'
+          PYTHON_VERSION: '3.9'
```
Can we maybe switch to 3.10 directly?

larsoner replied:
Not easily until VTK makes a wheel. They should release 9.2 soon, which will have one

hoechenberger commented on doc/install/advanced.rst:
```diff
@@ -191,6 +191,11 @@ to force MESA to use modern OpenGL by using this before executing
 
 Also, it's possible that different software rending backends might perform
 better than others, such as using the ``llvmpipe`` backend rather than ``swr``.
+In newer MESA (21+), rendering is off when using MSAA, so consider setting:
```
"off" as in "faulty"? Could this be rephrased to be more clear?

hoechenberger commented on doc/changes/latest.inc:
```diff
@@ -33,6 +33,8 @@ Bugs
 
 - Fix bug where ``theme`` was not handled properly in :meth:`mne.io.Raw.plot` (:gh:`10487`, :gh:`10500` by `Mathieu Scheltienne`_ and `Eric Larson`_)
 
+- Fix bug with newer MESA rendering made possible via ``MNE_3D_OPTION_MULTI_SAMPLES=1`` env var (:gh:`10513` by `Eric Larson`_)
```
Just reading this message, I didn't realize this variable was a **new** thing. How about:

```suggestion
- Rendering issues with recent MESA releases can be avoided by setting the new environment variable``MNE_3D_OPTION_MULTI_SAMPLES=1`` (:gh:`10513` by `Eric Larson`_)
```

hoechenberger replied:
But should probably also mention the new kwarg of `set_3d_options()`

larsoner commented:
Okay locally I was able to get PySide6 working by making appropriate use of `weakref`. Hopefully it comes back green!

@hoechenberger I addressed the comments above, so I'll merge if it does indeed come back green.

hoechenberger reviewed (commented):
LGTM! Thanks @larsoner, feel free to merge once green
